### PR TITLE
Make the timeout to consider workers offline configurable

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/69784#note-12

### CODE REVIEW

File: lib/OpenQA/Constants.pm [hunk 1]
```diff
@@ -25,8 +25,8 @@ use Exporter 'import';
 # If this value differs from server to worker then it won't be able to connect.
 use constant WEBSOCKET_API_VERSION => 1;
 
-# Time threshold used to check active workers (in seconds)
-use constant WORKERS_CHECKER_THRESHOLD => (2 * 24 * 60 * 60);
+# Default worker timeout
+use constant DEFAULT_WORKER_TIMEOUT => 1800;
```
[the remark this answers is not in the document]
Comment (Martchus) [reply lib/OpenQA/Constants.pm:30]: Ok, I'll split it in two commits.

File: lib/OpenQA/Log.pm [hunk 2]
```diff
@@ -73,7 +73,9 @@ sub log_fatal {
 
 sub _current_log_level {
     my $app = OpenQA::App->singleton;
-    return defined $app && $app->can('log') && $app->log->can('level') && $app->log->level;
+    return 0 unless defined $app && $app->can('log');
+    return 0 unless my $log = $app->log;
+    return $log->can('level') && $log->level;
```
[the remark this answers is not in the document]
Comment (Martchus) [reply lib/OpenQA/Log.pm:78]: It is required for the tests related to change. I could make it a separate commit which appears before the other commits, e.g. "Allow testing with an app which has no log".

[the remark this answers is not in the document]
Comment (okurz) [reply lib/OpenQA/Log.pm:78]: I see. Yes, please do that.

File: lib/OpenQA/Setup.pm [hunk 3]
```diff
@@ -195,6 +198,20 @@ sub read_config {
...
+}
+
+sub _validate_worker_timeout {
+    my ($app)                     = @_;
+    my $global_config             = $app->config->{global};
+    my $configured_worker_timeout = $global_config->{worker_timeout};
+    if (!looks_like_number($configured_worker_timeout) || $configured_worker_timeout < MAX_TIMER) {
+        $global_config->{worker_timeout} = DEFAULT_WORKER_TIMEOUT;
+        $app->log->warn(
+            'The specified worker_timeout is invalid and will be ignored. The timeout must be integer greather than '
```
Comment (kalikiana) [inline lib/OpenQA/Setup.pm:211]:
```suggestion
            'The specified worker_timeout is invalid and will be ignored. The timeout must be an integer greather than '
```

File: t/04-scheduler.t [hunk 4]
```diff
@@ -24,13 +24,15 @@ use OpenQA::Resource::Jobs;
...
 use OpenQA::Test::Database;
+use OpenQA::Test::Utils 'setup_mojo_app_with_default_worker_timeout';
 use OpenQA::Utils 'assetdir';
 use Test::Mojo;
 use Test::MockModule;
 use Test::Warnings ':report_warnings';
 use OpenQA::Schema::Result::Jobs;
 use OpenQA::WebAPI::Controller::API::V1::Worker;
 
+setup_mojo_app_with_default_worker_timeout;
```
Comment (kalikiana) [inline t/04-scheduler.t:35]: I wonder if this affects the reliability of this test - I was just investigating a case where `t_seen` was [off by a second in CircleCI](https://app.circleci.com/pipelines/github/os-autoinst/openQA/4196/workflows/bbc95dc4-cb0f-4426-82b0-826d3fce4848/jobs/40093).

Comment (Martchus) [reply t/04-scheduler.t:35]: In the failures you're referring `t_seen` has been updated unexpectedly. Not sure why that's the case but I doubt this line changes that.

**Review by okurz — APPROVED** [review]

**Comment by okurz** [comment]
https://app.circleci.com/pipelines/github/os-autoinst/openQA/4205/workflows/778e2876-2965-4426-bb28-fcd850fe900d/jobs/40210 shows

```
timeout -s SIGINT -k 5 -v $((10 * (3 + 1) ))m tools/retry prove -l --harness TAP::Harness::JUnit --timer --merge t/33-developer_mode.t
Retry 1 of 3 …
[16:28:14] t/33-developer_mode.t .. 25/? make[2]: *** [Makefile:174: test-unit-and-integration] Terminated
make[1]: *** [Makefile:169: test-with-database] Terminated
make: *** [Makefile:162: test-developer] Terminated

Too long with no output (exceeded 10m0s): context deadline exceeded
```

so similar to what @kalikiana was working on here we should also ensure that we have a time limit within 33-developer-mode.t which is less than 10m. I prepared https://github.com/os-autoinst/openQA/pull/3393 for this.

**Comment by kalikiana** [comment]
> https://app.circleci.com/pipelines/github/os-autoinst/openQA/4205/workflows/778e2876-2965-4426-bb28-fcd850fe900d/jobs/40210 shows
> 
> ```
> timeout -s SIGINT -k 5 -v $((10 * (3 + 1) ))m tools/retry prove -l --harness TAP::Harness::JUnit --timer --merge t/33-developer_mode.t
> Retry 1 of 3 …
> [16:28:14] t/33-developer_mode.t .. 25/? make[2]: *** [Makefile:174: test-unit-and-integration] Terminated
> make[1]: *** [Makefile:169: test-with-database] Terminated
> make: *** [Makefile:162: test-developer] Terminated
> 
> Too long with no output (exceeded 10m0s): context deadline exceeded
> ```
> 
> so similar to what @kalikiana was working on here we should also ensure that we have a time limit within 33-developer-mode.t which is less than 10m. I prepared #3393 for this.

And, too, addressing the potential cause for it: #3394

**Review by kalikiana — APPROVED** [review]

**Comment by okurz** [comment]
@kalikiana excellent :)

**Pull request merged**